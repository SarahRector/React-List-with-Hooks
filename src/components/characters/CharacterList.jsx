import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={`${character.name}-${character.image}`}>
      <Character {...character} />
    </li>
  ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
