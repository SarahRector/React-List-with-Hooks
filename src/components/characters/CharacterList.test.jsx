import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

const testArray = [
  { name: 'Arnold', image: 'test.png' },
  { name: 'Stevie', image: 'test2.png' } 
];

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders Character List', () => {
    const { asFragment } = render(<CharacterList
      characters={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
