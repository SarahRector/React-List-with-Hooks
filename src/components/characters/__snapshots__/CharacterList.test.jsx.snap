// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders Character List 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <div>
        <h1>
           Arnold 
        </h1>
        <img
          src="test.png"
        />
      </div>
    </li>
    <li>
      <div>
        <h1>
           Stevie 
        </h1>
        <img
          src="test2.png"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
